### PR TITLE
Add Transmission and Torrent objects

### PR DESCRIPTION
**Pull request by Krueladin**
Added Torrent object
Adds a simple representation of a torrent object. Currently not very functional and here mostly for future work.

Added TransmissionClient object
Adds functionality for interacting with a running Transmission Daemon.
- start_torrents
- stop_torrents
- get_torrents
- add_torrents
- add_torrent
- delete_torrents

### CODE REVIEW

File: transmission.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    pass
+
+class TransmissionClient(object):
+        def _post_request(self, data):
+            """Prepares the request to the Transmission client.
+
+            Sets the authentication for the client, as well as the data to be requested.
+            Resets the X-Transmission-Session-Id as needed.
+
+            data (json) -- A json representation of the data to be transmitted/requested.
```
Comment (zachary-trudo) [inline transmission.py]:
```
Args:
  data (json): A json rep ... requested.

Returns:
  response: The response from the transmission daemon
```

**Review by zachary-trudo — COMMENTED** [review]
Is there any way to go from filename to torrent id? I'd rather not create a list of torrent ids - but we can if need be.

**Comment by Krueladin** [comment]
What do you mean precisely?

**Comment by zachary-trudo** [comment]
I have a filename - I want to Update a torrent that has already been loaded. Can I hand transmission a filename and get the id in order to make these updates? If so can you add this functionality?

The other option is to record the ids as they are returned.

**Comment by Krueladin** [comment]
There is no obvious way to do that as far as I can tell. I could create a function that fetches all the torrents with their id and returns the torrent id where the filename matches.

File: transmission.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+            self.host = host
+            self.port = port
+            self.session_id = ""
+
+        # Action Requests 
+        def start_torrents(self, t_ids=[]):
+            """Starts the torrent files of the associated ids passed in.
+
+            Args:
+            t_ids ([]int): The array of torrent ids to begin torrenting.
```
Comment (zachary-trudo) [inline transmission.py:57]: This is python not stupid go... 
list or [] is fine.
if you really need to... [int]

**Comment by zachary-trudo** [comment]
LGTM - should be easy to implement most functionality of the adapter here.

**Pull request merged**